x-pack/legacy/plugins/code/public/components/main/top_bar.tsx

```diff
@@ -25,21 +25,50 @@ interface Props {
 
 export class TopBar extends React.Component<Props, { value: string }> {
   public state = {
-    value: 'master',
+    value: decodeRevisionString(this.props.routeParams.revision),
   };
 
-  public onChange = (e: ChangeEvent<HTMLSelectElement>) => {
+  get branch() {
+    return this.getBranch(this.state.value);
+  }
+
+  getBranch = (revision: string) => {
+    const r = decodeRevisionString(revision);
+    const branch = this.props.branches.find(b => b.name === r);
+    const sameCommitBranch = this.props.branches.find(b => b.commit.id.startsWith(r));
```

Comment:
This is a thoughtful implementation. However, for any given revision, which does not match any branches, we need some sort of design here. @daveyholler GitHub has the following design:

![image](https://user-images.githubusercontent.com/987855/59951944-5e6baf00-942f-11e9-8737-aad3724d281e.png)

While currently with this PR, it does not change when there is a revision in the URL
![image](https://user-images.githubusercontent.com/987855/59951888-27959900-942f-11e9-8eb5-eca8bda1e59d.png)

Reply:
set `select` value to empty `` for now